core/src/apps/bitcoin/sign_tx/scripts.py

```diff
@@ -89,13 +86,13 @@ def output_derive_script(txo: TxOutputType, coin: CoinInfo) -> bytes:
         elif version == cashaddr.ADDRESS_TYPE_P2SH:
             version = coin.address_type_p2sh
         else:
-            raise ScriptsError("Unknown cashaddr address type")
```

Comment:
I believe this was a bug (missing FailureType). Occurs in other places in scripts.py and decred.py.

Reply:
agreed. that also shows that we don't have test coverage for this.
(maybe it doesn't really matter though?)

Reply:
Indeed. Honestly I was wondering if you will suggest to write the test yourself :). We can do it, but writing test for each failing condition seems a bit annoying and maybe indeed redundant.